bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/http/HttpResponseListener.java

```diff
@@ -0,0 +1,93 @@
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HttpResponseListener} is responsible for processing the result of a HTTP request
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class HttpResponseListener extends BufferingResponseListener {
+    private final Logger logger = LoggerFactory.getLogger(HttpResponseListener.class);
+    private final CompletableFuture<@Nullable Content> future;
+    private final String fallbackEncoding;
+
+    public HttpResponseListener(CompletableFuture<@Nullable Content> future) {
+        this(future, null);
+    }
+
+    public HttpResponseListener(CompletableFuture<@Nullable Content> future, @Nullable String fallbackEncoding) {
+        this.future = future;
+        this.fallbackEncoding = fallbackEncoding != null ? fallbackEncoding : StandardCharsets.UTF_8.name();
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void onComplete(@NonNullByDefault({}) Result result) {
+        Response response = result.getResponse();
+        if (logger.isTraceEnabled()) {
+            logger.trace("Received from '{}': {}", result.getRequest().getURI(), responseToLogString(response));
+        }
+        Request request = result.getRequest();
+        if (result.isFailed()) {
+            logger.warn("Requesting '{}' (method='{}', content='{}') failed: {}", request.getURI(), request.getMethod(),
+                    request.getContent(), result.getFailure().getMessage());
+            future.complete(null);
+        } else {
+            switch (response.getStatus()) {
+                case HttpStatus.OK_200:
+                    byte[] content = getContent();
+                    String encoding = getEncoding();
+                    if (content != null) {
+                        future.complete(
+                                new Content(content, encoding == null ? fallbackEncoding : encoding, getMediaType()));
+                    } else {
+                        future.complete(null);
+                    }
+                    break;
+                case HttpStatus.UNAUTHORIZED_401:
+                    logger.debug("Requesting '{}' (method='{}', content='{}') failed: Authorization error",
+                            request.getURI(), request.getMethod(), request.getContent());
+                    future.completeExceptionally(new HttpAuthException());
+                    break;
+                default:
+                    logger.info("Requesting '{}' (method='{}', content='{}') failed: {} {}", request.getURI(),
```

Comment:
`info` is not adequate here. See https://www.openhab.org/docs/developer/guidelines.html#f-logging

"info logging should be used sparsely. e.g. a newly started component or a user file that has been loaded."

Reply:
Since this is an error state, I raised it to warn.